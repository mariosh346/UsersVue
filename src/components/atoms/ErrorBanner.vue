<template>
  <q-banner
    v-if="message"
    inline-actions
    class="text-white bg-red"
  >
    {{ message }}
  </q-banner>
</template>

<script setup lang="ts">
defineProps<{
  message?: string;
}>();
</script>
