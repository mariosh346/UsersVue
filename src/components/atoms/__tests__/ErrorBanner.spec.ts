import { describe, it, expect, beforeEach } from 'vitest';
import { shallowMount } from '@vue/test-utils';
import ErrorBanner from "../ErrorBanner.vue";

describe('ErrorBanner', () => {
  let wrapper: ReturnType<typeof shallowMount>;
  const message = 'Test error';
  const createWrapper = () => shallowMount(ErrorBanner, {
    props: { message }
  });

  beforeEach(() => {
    wrapper = createWrapper();
  });

  it('matches snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
