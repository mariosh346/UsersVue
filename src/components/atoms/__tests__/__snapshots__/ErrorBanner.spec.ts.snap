// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ErrorBanner > matches snapshot 1`] = `"<q-banner-stub inlineactions="true" dense="false" rounded="false" class="text-white bg-red"></q-banner-stub>"`;
